File: frontend/lib/api/payment.ts
import { apiClient } from './client'

export interface Payment {
  id: number
  invoice_id: number
  amount: number
  payment_method: string // cash, telebirr, bank, other
  bank_name?: string
  payment_date: string
  notes?: string
  created_by?: number
  created_at: string
}

export interface PaymentCreate {
  invoice_id: number
  amount: number
  payment_method: string
  bank_name?: string
  payment_date?: string
  notes?: string
}

export interface PaymentSummary {
  invoice_id: number
  invoice_total: number
  total_paid: number
  outstanding_balance: number
  status: string
}

export const paymentApi = {
  createPayment: async (data: PaymentCreate): Promise<Payment> => {
    const response = await apiClient.post('/payment', data)
    return response.data
  },

  getInvoicePayments: async (invoiceId: number): Promise<Payment[]> => {
    const response = await apiClient.get(`/payment/invoice/${invoiceId}`)
    return response.data
  },

  getInvoicePaymentSummary: async (invoiceId: number): Promise<PaymentSummary> => {
    const response = await apiClient.get(`/payment/invoice/${invoiceId}/total`)
    return response.data
  },
}

File: backend/app/schemas/payment.py
"""
Payment schemas
"""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class PaymentCreate(BaseModel):
    invoice_id: int
    amount: float
    payment_method: str = "cash"  # cash, telebirr, bank, other
    bank_name: Optional[str] = None
    payment_date: Optional[datetime] = None
    notes: Optional[str] = None


class PaymentResponse(BaseModel):
    id: int
    invoice_id: int
    amount: float
    payment_method: str
    bank_name: Optional[str] = None
    payment_date: datetime
    notes: Optional[str] = None
    created_by: Optional[int] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

